src/routes/turnilo_dashboard_routes.py

```diff
@@ -0,0 +1,85 @@
+import re
+from fastapi import APIRouter, Depends, HTTPException, Query
+from typing import List
+from sqlmodel import Session
+from models.turnilo_dashboard import TurniloDashboard
+from services import turnilo_dashboards as td
+
+import constants
+import data.database as db
+
+turnilo_router = APIRouter()
+
+### Dashboards ###
+
+
+def is_valid_query(s: str) -> bool:
+    if not isinstance(s, str):
```

Comment:
After digging on the FastAPI docs, I think there is no need to check types, as they always be converted on the internal dependency chain and provided to the controller function or return queryError. 

Related to the length and pattern. FastApi let's you declare that in the query definition:
``` 
Query(          
            title="Query string",
            description="Query string for the items to search in the database that have a good match",
            min_length=3,
            max_length=50,
            pattern="^fixedquery$",
```
Although it is way more verbose, I personally don't like it, the advantage is that all of this will be added to the openAPI spec which will enrich the overall documentation of the API

Reply:
Additionally, my preferred way would be to use pydantic models as the Query Drivers, They would be minimal, easy to document, typed, you can add additional validation, And you can use inheritance on them. Moreover, they look just a as single param in the functions ensuring minimal boilerplate and easy maintenance on the HTTP controllers, example: 

```python
from fastapi import FastAPI, Depends
from pydantic import BaseModel, Field
from typing import Optional

app = FastAPI()

class QueryParams(BaseModel):
    """
    Class representing the query params, there are 3 of them with types str, int and bool where
    the last one is optional
    """
    param1: str = Field(..., examples=["value1"], description="A required string parameter")
    param2: int = Field(..., examples=[123], description="A required integer parameter")
    param3: Optional[bool] = Field(None, examples=[True], description="An optional boolean parameter")

@app.get("/items/", summary="Get items", description="Retrieve items with the given query parameters")
async def read_items(query_params: QueryParams = Depends()):
    """
    This endpoint retrieves items based on the provided query parameters.
    - **param1**: A required string parameter
    - **param2**: A required integer parameter
    - **param3**: An optional boolean parameter
    """
    return {"param1": query_params.param1, "param2": query_params.param2, "param3": query_params.param3}

if __name__ == "__main__":
    import uvicorn
    uvicorn.run(app, host="", port=8000)
```

after running this and calling it:

```
http://localhost:8000/items/?param1=exq&param2=123&param3=true&noexist=aaa
```
Mind the `noexist` param, you will get:

```json
{"param1":"exq","param2":123,"param3":true}
```
and of course, they also appear on the API documentation

![image](https://github.com/datahangar/backend/assets/1920980/ca56810d-ff82-44e9-b5b8-13e8f6a6b9e0)

Reply:
The option I gave I think is way more extensible and easy to re-use and maintain, because if you ever need to put pagination in the routers, you will have to add and change all the signatures of the functions

Reply:
Let's go with your wise advise:

1. Remove type validators
2. Go with the Query obj.

Give me some time to push it.

Reply:
Applied; I couldn't find a way to have the `QueryParams` class in the routes module, instead of the services, due to a circulate dependency.

I think it's kind of unavoidable, due to typing... I cannot import it in the function signature. But if you know how to do it, lemme know :rocket:

Reply:
I don't think is a big deal and I liked were thinks stand, as a workaround you can annotate the function with "" instead of the class type. But is usually advisable not to have circular dependencies even if it is for annotation purtposes